fix: Disable debug trace

dupdob opened the pull request:
Make Stryker compatible with debug assertions logic, so that a mutant is considered as killed if it result in a failed assertion.

Disable `Debug.Assert` report dialogs. Failed assertions result in an argument exception raised, ensuring this are recognized as failed tests by Stryker.

Fixes #2375

Note: coverage reporting is wrong. For some reason, Sonar does not capture any coverage data for the DataCollector. I have added a dedicated test that does cover all testable lines.

Liam-Rougoor commented:
/azp run

Liam-Rougoor reviewed (commented):
Thanks for the PR, it looks good. I do think this case could benefit from an integration test as well.

dupdob commented:
I added an integration test, see [here](https://github.com/stryker-mutator/stryker-net/blob/09243bc123b5461c8625f47b640afaf66d8b0ef1/integrationtest/TargetProjects/NetCore/NetCoreTestProject.XUnit/VariousTests.cs#L7) and adjusted test expectations accordingly.
